File: src/main/java/com/example/chatService/entity/RoomBan.java
package com.example.chatService.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Entity
@Table(
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"roomId", "userId"})
        }
)
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RoomBan {

    @Id
    @GeneratedValue
    private Long id;

    private String roomId;
    private String userId;

    private String bannedBy; // OWNER
    private OffsetDateTime bannedAt;
}

File: src/main/java/com/example/chatService/entity/RoomBan.kt
package com.example.chatService.entity

import jakarta.persistence.*
import java.time.OffsetDateTime

@Entity
@Table(
    uniqueConstraints = [
        UniqueConstraint(columnNames = ["roomId", "userId"])
    ]
)
class RoomBan(
    @Id
    @GeneratedValue
    var id: Long? = null,
    var roomId: String? = null,
    var userId: String? = null,
    var bannedBy: String? = null,
    var bannedAt: OffsetDateTime? = null
)
